.github/workflows/test.yaml

```diff
@@ -164,8 +164,19 @@ jobs:
           # Start a TryCP instance
           nix develop .#ci -c bash -c "source ./scripts/trycp.sh && start_trycp"
 
-          # Run the scenario for 30 seconds
-          RUST_LOG=info nix run .#remote_call_rate -- --targets targets-ci.yaml --instances-per-target 2 --duration 30 --no-progress
+          # Run the scenario for 10 seconds
+          RUST_LOG=warn MIN_PEERS=2 nix run .#remote_call_rate -- --targets targets-ci.yaml --instances-per-target 2 --duration 10 --no-progress
```

Comment:
Is 30 seconds too long?

Reply:
I think so. Unless we assert the results, this is just about getting the code to execute at least once. it doesn't need to run for long.

I'm thinking about adding a 'max iterations' or something like that to the CLI flags for the runner. That way we could set the timeout much higher and limit how many times the behaviour runs to something like 3 times. That would eliminate guessing on CI and ensure that we always test the agent behaviour even when things run slowly. It would also be really useful for debugging. When something goes wrong with a scenario it can really spam errors. If we could limit the number of iterations then it would be easier to stop the scenario and look at the first few errors.